api/server/router/container/container.go

```diff
@@ -60,6 +60,7 @@ func (r *containerRouter) initRoutes() {
 		// PUT
 		local.NewPutRoute("/containers/{name:.*}/archive", r.putContainersArchive),
 		// DELETE
+		local.NewDeleteRoute("/containers/_all", r.deleteAllContainers),
```

Comment:
I don't like `_all`, I would prefer the `/containers` endpoint. There is a question of whether it would also work with `/containers/` in which case an empty value for `$name` in `/containers/$name` would inadvertently result in deleting all. If we could return a well phrased error on `/containers/` and use the `/containers` endpoint for deleting all containers, that would be my preferred.

I am aware there is no well defined convention for how to treat an endpoint with "trailing slash" vs without trailing slash. Clients and proxies should be fine as far as I can tell, since `/foo` and `/foo/` are two different URIs and it's always the server's responsibility to do or not do a redirect.

If the only issue with my proposal is that it's not a common thing to do, I still think we should do it.